src/unnest.c

```diff
@@ -0,0 +1,226 @@
+#include "data.table.h"
+#include <Rdefines.h>
+
+SEXP unnest(SEXP x) {
+  int n = LENGTH(VECTOR_ELT(x, 0));
+  int p = LENGTH(x);
+  int row_counts[n];
+  SEXPTYPE col_types[p];
```

Comment:
Could segfault for many rows or columns

Reply:
Would also be obviated by switch to `cols`...

The issue here is if we try to instantiate `SEXPTYPE` array with `>INT_MAX` elements?